### ISSUE TITLE
rocsparse spsv analysis phase 

### ISSUE THREAD

**Issue by uesrshihe** [ROCmSoftwarePlatform/rocSPARSE#424]
"I have a question. In the rocsparse SPSV (Sparse Solver), there are three stages, and the second stage is the analysis phase, which generates the dependency array done_array. I am trying to think about how each element in the done_array is generated outside of the code logic. I converted the sparse matrix into a DAG (Directed Acyclic Graph), and found that the number of dependencies for each row does not match the values in the done_array. So, outside of the code, what is the basis for generating the done_array array?"

**Comment by ntrost57** [comment]
Hi uesrshihe, it is not clear to me what you are asking for. Could you rephrase your question?

**Comment by uesrshihe** [comment]
I have a question about the rocsparse SPSV (Sparse Solver). In the analysis phase (the second stage), a dependency array called done_array is generated. I am trying to understand how each element of this array is computed outside the code logic. I converted the sparse matrix into a Directed Acyclic Graph (DAG), but I noticed that the number of dependencies per row in the DAG does not match the corresponding values in the done_array.

From my understanding, the value for each node in done_array corresponds to the maximum path from that node to another node in the DAG that has no dependencies. Is this correct? If not, could you explain how the dependencies are defined and calculated during the analysis phase?

**Comment by ntrost57** [comment]
What do you mean by outside the code logic? You can see how the done array is generated [here](https://github.com/ROCm/rocSPARSE/blob/063ee73e3b2d78c22eb7890e5c88ad90600adc3c/library/src/level2/csrsv_device.h#L35)

**Comment by ammallya** [comment]
Imported to ROCm/rocm-libraries
